data/templates/snmp/etc.snmpd.conf.j2

```diff
@@ -59,27 +59,71 @@ agentaddress unix:/run/snmpd.socket{{ ',' ~ options | join(',') if options is vy
 {%             for client in comm_config.client %}
 {%                 if client | is_ipv4 %}
-{{ comm_config.authorization }}community {{ comm }} {{ client }}
+{{ comm_config.authorization }}community {{ comm }} {{ client }} -V RESTRICTED
 {%                 elif client | is_ipv6 %}
-{{ comm_config.authorization }}community6 {{ comm }} {{ client }}
+{{ comm_config.authorization }}community6 {{ comm }} {{ client }} -V RESTRICTED
 {%                 endif %}
 {%             endfor %}
 {%         endif %}
 {%         if comm_config.network is vyos_defined %}
 {%             for network in comm_config.network %}
 {%                 if network | is_ipv4 %}
-{{ comm_config.authorization }}community {{ comm }} {{ network }}
+{{ comm_config.authorization }}community {{ comm }} {{ network }} -V RESTRICTED
 {%                 elif network | is_ipv6 %}
-{{ comm_config.authorization }}community6 {{ comm }} {{ network }}
+{{ comm_config.authorization }}community6 {{ comm }} {{ network }} -V RESTRICTED
 {%                 endif %}
 {%             endfor %}
 {%         endif %}
 {%         if comm_config.client is not vyos_defined and comm_config.network is not vyos_defined %}
-{{ comm_config.authorization }}community {{ comm }}
+{{ comm_config.authorization }}community {{ comm }} -V RESTRICTED
+{{ comm_config.authorization }}community6 {{ comm }} -V RESTRICTED
 {%         endif %}
 {%     endfor %}
 {% endif %}
 
+# Default RESTRICTED view
+view RESTRICTED    included .1 80
+{% if oid_enable is not vyos_defined %}
+## No oid-enabed set. Disable all OID as stands FRR docs
+# ipRouteTable oid: excluded
+view RESTRICTED    excluded  .1.3.6.1.2.1.4.21
+# ipNetToMediaTable oid: excluded
+view RESTRICTED    excluded  .1.3.6.1.2.1.4.22
+# ipNetToPhysicalPhysAddress oid: excluded
+view RESTRICTED    excluded  .1.3.6.1.2.1.4.35
+# ipCidrRouteTable oid: excluded
+view RESTRICTED    excluded  .1.3.6.1.2.1.4.24
+{% elif 'all' not in oid_enable %}
+{%     if 'ip-route-table' in oid_enable %}
+# ipRouteTable oid: included
+view RESTRICTED    included  .1.3.6.1.2.1.4.21
+{%     else %}
+# ipRouteTable oid: excluded
+view RESTRICTED    excluded  .1.3.6.1.2.1.4.21
+{%     endif %}
+{%     if 'ip-net-to-media-table' in oid_enable %}
+# ipNetToMediaTable oid: included in view
+view RESTRICTED    included  .1.3.6.1.2.1.4.22
+{%     else %}
+# ipNetToMediaTable oid: excluded
+view RESTRICTED    excluded  .1.3.6.1.2.1.4.22
+{%     endif %}
+{%     if 'ip-net-to-physical-phys-address' in oid_enable %}
+# ipNetToPhysicalPhysAddress oid: included
+view RESTRICTED    included  .1.3.6.1.2.1.4.35
+{%     else %}
+# ipNetToPhysicalPhysAddress oid: excluded
+view RESTRICTED    excluded  .1.3.6.1.2.1.4.35
+{%     endif %}
+{%     if 'ip-cidr-route-table' in oid_enable %}
+# ipCidrRouteTable oid: included
+view RESTRICTED    included  .1.3.6.1.2.1.4.24
+{%     else %}
+# ipCidrRouteTable oid: excluded
+view RESTRICTED    excluded  .1.3.6.1.2.1.4.24
+{%     endif %}
+{% endif %}
```

Comment:
Most likely, without the `all` option, this can be simplified to:
```
# Default RESTRICTED view
view RESTRICTED    included .1 80
{% if oid_enable.ip_route_table is not vyos_defined %}
# ipRouteTable oid: excluded
view RESTRICTED    excluded  .1.3.6.1.2.1.4.21
{% endif %}
```
And so on.
I am only not sure if the `-` and `_` in options are automatically converted.